### ISSUE TITLE
Hardware breakpoints are not applied when enabling them during system breakpoint

### ISSUE THREAD

**Issue by dauthleikr** [x64dbg/x64dbg#2297]
- Debugger version: Feb23 2020
- Operating system version: Win10 1909 (18363.657)
- No plugins, fresh download

x64dbg does not set hardware breakpoints (even though it says it did) when **re-enabling** one during system **breakpoint**. 

Steps to reproduce:
0. Have System Breakpoint enabled in Settings -> Events
1. Start an application with x64dbg
2. Place a hardware breakpoint somewhere where it will be triggered (e.g. somewhere in main)
3. Disable the hardware breakpoint
4. Restart the app with x64dbg using the restart button at the top left
5. As soon as x64dbg runs into the system breakpoint at the start, go to breakpoints and re-enable your hardware breakpoint. Note that x64dbg says "Hardware breakpoint enabled!" and the breakpoint is listed as enabled
6. Press the "Run" button at the top left to continue execution
7. Note how the program does not run into the hardware breakpoint, even though it is supposedly enabled

Using `GetThreadContext` to get Dr0-Dr3 confirms that the debug registers are zero, and the breakpoint has not been placed,

**Comment by mrexodia** [comment]
I believe this is a known issue and a limitation in the way Windows works, because at the system breakpoint the debug registers are reset no matter what.

**Comment by dauthleikr** [comment]
In that case maybe x64dbg should check if the process is currently paused at a system breakpoint when enabling hardware breakpoints, and print an appropriate message (e.g. "can not set hw bp during system breakpoint") and not display them as enabled, because the "Hardware breakpoint enabled!" message and their state in the breakpoint tab can be quite misleading.

**Comment by mrexodia** [comment]
Yeah unfortunately this is not so easy to do, but worth looking into I
guess.

**Comment by VladlenaVashchenko** [comment]
Dear mrexodia,

How are you love?

I have the same issue and I checked your latest build in win 10 and win 7. HWBP works great on windows 7, it breaks only in windows 10. Olly works fine in all windows. Would you please check it. Only you can fix it, all it takes is your interest my dear.

Regard,
Vivi
